bootstrap/httpserver/server_handlers_test.go

```diff
@@ -222,6 +223,51 @@ func TestHttpServer_Robots(t *testing.T) {
 	})
 }
 
+func TestPublicApiResponds503UntilPublicApiIsRegistered(t *testing.T) {
+	with.Logging(t, func(harness *with.LoggingHarness) {
+		h := newHarnessWithUnregisteredPublicApi(harness)
+		defer h.server.Shutdown()
+		port := h.server.Port()
```

Comment:
since you already have a harness, you could push the http request logic, including crafting the url, into the harness